### DISCUSSION TITLE
How can the Versatile Thermostat setup be smart enough to well take into account off-peak or full jours OR/AND free energy that come from a photovoltaic system ?

### DISCUSSION THREAD

**Discussion by afaucogney** [jmcollin78/versatile_thermostat#187] [General]
Hello,

Is there a smart strategy to setup the thermostat to handle a better energy cost management ?
- I have a PV system that produce free/cheaper electrical energy, when sun is available and when I do not use it for something else
- I have a off-peak / full peak (french EDF Tempo) contract, with 1 slot during the day, where the energy is cheaper, and 3 types of days ! Where the days (RED) are very expensive during the full hours = I must avoid to warm during this period.
  - My heaters (I have 5), based on Head Pump, are warming inside the floor, so the inertia is high. So I would like to warm enough the floor during the night, to get a good temperature during the day, with a min of energy consumption during the day !

I know I ask a lot, I know and I didn't see in the doc this way of expectation, but if someone has try some side strategy that goes in my direction, that will definitively help me ! and maybe others ....

**Comment by jmcollin78** [comment]
Hello @afaucogney ,

I've got the same configuration (PV and EDF Tempo). All is possible simply is to force switching to a higher preset (Boost) when power is available. The problem with that approach, is that you will be warm when you don't need it (during the day).

This could be done with an integration but also with this integration with is able to switch some equipments when power is available. You can have a look if you think it is relevant: https://github.com/jmcollin78/solar_optimizer

I think the only accurate way, is to have batteries which will be able to give some power when you need it (morning, evening and night). Especially in winter when less sun is available. With tempo you can fulfill your batteries at night and use it the day.

This is a great discusion and thank you for that ! I'm curious of the solutions the community have to this problem.

**Comment by AlexandreLine** [comment]
Hi @jmcollin78 

First at all, I wanted to thank you for your amazing work on this package. I just discovered it, and hope to automate and save some electricity this winter better my electrical heaters.

As you and @afaucogney , I'm running under EDF tempo contract. And I expected to run the same strategy described above warming house using cheaper electricity and slowly decrease house temperature during expensive times running on inertia.

Were you able to make some tests regarding this strategy ? I suppose it may vary depending on house insulation... But I would very much appreciate feedbacks from learned experience and experts !

Thanks in advance for your answers.
Alex

**Reply by AlexandreLine** [reply]
In addition, would it be easy to schedule a positive or négative temperature slope target (ie:from 23°C at 6:00 to 18°C at 22:00 ) instead of a fix temperature target?

**Reply by jmcollin78** [reply]
Slope is not very reliable information. Only the sign can interesting to look at.
